modules/openapi-generator/src/test/java/org/openapitools/codegen/go/GoClientCodegenTest.java

```diff
@@ -222,6 +222,27 @@ public void verifyTestFile() throws IOException {
                 "func Test_openapi_PetApiService(t *testing.T) {");
     }
 
+    @Test
+    public void verifyTestImport() throws IOException {
+        File output = Files.createTempDirectory("test").toFile();
+        output.deleteOnExit();
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("go")
+                .setGitUserId("OpenAPITools")
+                .setGitRepoId("openapi-generator")
+                .setInputSpec("src/test/resources/3_0/petstore.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(configurator.toClientOptInput()).generate();
+        files.forEach(File::deleteOnExit);
+
+        TestUtils.assertFileExists(Paths.get(output + "/test/api_pet_test.go"));
+        TestUtils.assertFileContains(Paths.get(output + "/test/api_pet_test.go"),
+                "openapiclient \"github.com/OpenAPITools/openapi-generator\"");
+    }
+
```

Comment:
Out of curiosity, is there some test we can write to generate a test client and run this test in Go? Trying to think of how we can avoid an issue like this in the future.

Reply:
this is because the test files are not overwritten by default.

moving forward, for changes in the api test template, we should remove the auto-generated test files manually (assuming these test files do not contain manually written test cases) so as to catch issue like this one.